aten/src/ATen/native/ConvUtils.h

```diff
@@ -425,25 +425,16 @@ inline bool xpu_conv_use_channels_last(const at::Tensor& input, const at::Tensor
   if (!input.is_xpu() || !weight.is_xpu()) {
     return false;
   }
-
-  // disable NHWC for float64 input.
-  if (input.scalar_type() == at::kDouble ||
-      weight.scalar_type() == at::kDouble) {
+  if (!input.defined() || input.is_sparse()) {
+    // suggest channels_first
```

Comment:
@ZhiweiYan-96 , has the XPU supported double?

Reply:
XPU has double support for `conv_fwd, conf_bwd_data`, but no support for `conv_bwd_weight`